pkg/rke2/config.go

```diff
@@ -307,17 +309,40 @@ type rke2AgentConfig struct {
 }
 
 type RKE2AgentConfigOpts struct {
-	ServerURL   string
-	Token       string
-	AgentConfig bootstrapv1.RKE2AgentConfig
-	Ctx         context.Context
-	Client      client.Client
+	ServerURL              string
+	Token                  string
+	AgentConfig            bootstrapv1.RKE2AgentConfig
+	Ctx                    context.Context
+	Client                 client.Client
+	CloudProviderName      string
+	CloudProviderConfigMap *corev1.ObjectReference
 }
 
 func newRKE2AgentConfig(opts RKE2AgentConfigOpts) (*rke2AgentConfig, []bootstrapv1.File, error) {
 	rke2AgentConfig := &rke2AgentConfig{}
 	files := []bootstrapv1.File{}
 	rke2AgentConfig.ContainerRuntimeEndpoint = opts.AgentConfig.ContainerRuntimeEndpoint
+	if opts.CloudProviderConfigMap != nil {
+		cloudProviderConfigMap := &corev1.ConfigMap{}
+		if err := opts.Client.Get(opts.Ctx, types.NamespacedName{
+			Name:      opts.CloudProviderConfigMap.Name,
+			Namespace: opts.CloudProviderConfigMap.Namespace,
+		}, cloudProviderConfigMap); err != nil {
+			return nil, nil, fmt.Errorf("failed to get cloud provider config map: %w", err)
+		}
+		cloudProviderConfig, ok := cloudProviderConfigMap.Data["cloud-config"]
+		if !ok {
+			return nil, nil, fmt.Errorf("cloud provider config map is missing cloud-config key")
+		}
+		rke2AgentConfig.CloudProviderConfig = "/etc/rancher/rke2/cloud-provider-config"
+		files = append(files, bootstrapv1.File{
+			Path:        rke2AgentConfig.CloudProviderConfig,
+			Content:     cloudProviderConfig,
+			Owner:       "root:root",
```

Comment:
We have constant for these elsewhere, it would be worth moving them to a shared "constants" package.

Reply:
What do you think about a `consts.go` in the `util` package?

Reply:
Or a separate constants package? Its up to you, main thing is it would be good to have these defined once.

Reply:
Tracking this through issue #81.